core/common/test/samples/format/DateTimeFormatSamples.kt

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2019-2024 JetBrains s.r.o. and contributors.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.datetime.test.samples.format
+
+import kotlinx.datetime.*
+import kotlinx.datetime.format.*
+import kotlin.test.*
+
+class DateTimeFormatSamples {
+
+    @Test
+    fun format() {
+        // Formatting a date using a predefined format
+        check(LocalDate.Formats.ISO.format(LocalDate(2021, 2, 7)) == "2021-02-07")
+    }
+
+    @Test
+    fun formatTo() {
+        // Appending a formatted date to an `Appendable` (e.g. a `StringBuilder`)
+        val sb = StringBuilder()
+        sb.append("Today is ")
+        LocalDate.Formats.ISO.formatTo(sb, LocalDate(2024, 4, 5))
+        check(sb.toString() == "Today is 2024-04-05")
+    }
+
+    @Test
+    fun parse() {
+        // Parsing a string that is expected to be in the given format
+        check(LocalDate.Formats.ISO.parse("2021-02-07") == LocalDate(2021, 2, 7))
+        try {
+            LocalDate.Formats.ISO.parse("2021-02-07T")
+            fail("Expected IllegalArgumentException")
+        } catch (e: IllegalArgumentException) {
+            // the input string is not in the expected format
+        }
+        try {
+            LocalDate.Formats.ISO.parse("2021-02-40")
+            fail("Expected IllegalArgumentException")
+        } catch (e: IllegalArgumentException) {
+            // the input string is in the expected format, but the value is invalid
+        }
+        // to parse strings that have valid formats but invalid values, use `DateTimeComponents`:
+        check(DateTimeComponents.Format { date(LocalDate.Formats.ISO) }.parse("2021-02-40").dayOfMonth == 40)
+    }
+
+    @Test
+    fun parseOrNull() {
+        // Attempting to parse a string that may not be in the expected format
+        check(LocalDate.Formats.ISO.parseOrNull("2021-02-07") == LocalDate(2021, 2, 7))
+        check(LocalDate.Formats.ISO.parseOrNull("2021-02-07T") == null)
+        check(LocalDate.Formats.ISO.parseOrNull("2021-02-40") == null)
+        check(LocalDate.Formats.ISO.parseOrNull("2021-02-40") == null)
+        // to parse strings that have valid formats but invalid values, use `DateTimeComponents`:
+        val dateTimeComponentsFormat = DateTimeComponents.Format { date(LocalDate.Formats.ISO) }
+        check(dateTimeComponentsFormat.parseOrNull("2021-02-40")?.dayOfMonth == 40)
+    }
+
+    @Test
+    fun formatAsKotlinBuilderDsl() {
+        // Printing a given date-time format as a Kotlin code snippet that creates the same format
+        val customFormat = LocalDate.Format {
+            @OptIn(FormatStringsInDatetimeFormats::class)
+            byUnicodePattern("MM/dd uuuu")
```

Comment:
Would be cool to show the substitution for `yyyy` which is more widespread instead.

Reply:
Done.